src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java

```diff
@@ -1456,20 +1475,22 @@ public void requestAndShowCodeActions() {
                         int startOffset = editor.getDocument().getLineStartOffset(line);
                         int endOffset = editor.getDocument().getLineEndOffset(line);
                         TextRange range = new TextRange(startOffset, endOffset);
-
-                        try {
-                            this.anonHolder
-                                    .newAnnotation(HighlightSeverity.INFORMATION, codeAction.getTitle())
-                                    .range(range)
-                                    .withFix(new LSPCodeActionFix(FileUtils.editorToURIString(editor), codeAction))
-                                    .create();
-
-                            SmartList<Annotation> asList = (SmartList<Annotation>) this.anonHolder;
-                            this.annotations.add(asList.get(asList.size() - 1));
-                            diagnosticSyncRequired = true;
-                        } catch (IllegalArgumentException e) {
-                            LOG.warn("Error when creating a new annotation");
+                        Tuple3<HighlightSeverity, TextRange, LSPCodeActionFix> triple =
+                                new Tuple3<>(
+                                        HighlightSeverity.INFORMATION,
+                                        range,
+                                        new LSPCodeActionFix(FileUtils.editorToURIString(editor), codeAction)
+                                );
+                        boolean found = silentAnnotations.stream()
+                                .anyMatch(silentAnnotation ->
+                                        silentAnnotation.getSecond().getStartOffset() == startOffset &&
+                                        silentAnnotation.getSecond().getEndOffset() == endOffset &&
+                                        silentAnnotation.getThird().getText().equals(codeAction.getTitle())
+                                 );
+                        if (!found) {
+                            silentAnnotations.add(triple);
```

Comment:
We can move the silent annotation object creation (L1478) into this code block, so that the object gets created only when the condition is false